include/alpaka/core/Cuda.hpp

```diff
@@ -177,15 +177,31 @@ namespace alpaka
                 int const & line)
             -> void
             {
-                if(error != CUDA_SUCCESS)
-                {
-                    std::string const sError(std::string(file) + "(" + std::to_string(line) + ") " + std::string(desc) + " : '" + cudaGetErrorName(static_cast<cudaError_t>(error)) +  "': '" + std::string(cudaGetErrorString(static_cast<cudaError_t>(error))) + "'!");
+                if(error == CUDA_SUCCESS)
+                    return;
+
+                char const * cu_err_string = nullptr;
+                char const * cu_err_name = nullptr;
+                CUresult cu_result_name = cuGetErrorName(error, &cu_err_name);
+                CUresult cu_result_string = cuGetErrorString(error, &cu_err_string);
+                std::string sError;
+                if( cu_result_name == CUDA_ERROR_INVALID_VALUE
+                    || cu_result_string == CUDA_ERROR_INVALID_VALUE ) {
+                    sError = std::string(__FILE__)
+                        + "(" + std::to_string(__LINE__) + ") "
+                        + "CUDA Driver API: Invalid Value.";
+                } else {
+                    sError = std::string(file)
+                        + "(" + std::to_string(line) + ") "
```

Comment:
You could extract the duplicate `sError = std::string(file) + "(" + std::to_string(line) + ") "` diretly to the declaration of the string in front of the if.